language/move-vm/types/src/natives/function.rs

```diff
@@ -65,15 +65,22 @@ pub struct NativeResult {
     /// The cost for running that function, whether successfully or not.
     pub cost: GasUnits<GasCarrier>,
     /// Result of execution. This is either the return values or the error to report.
-    pub result: Result<Vec<Value>, u64>,
+    pub result: Result<Option<Value>, u64>,
```

Comment:
this (and the below `ok_one` and `ok_none`) makes a lot of sense. I think the way we may want to steal it is with an enum that has 0, 1 and multiple return values. The way this is now is not quite right given a function may return multiple values. However I suppose they all return `void` or 1 value now and it makes sense to have a variant for void, 1 or multiple returns. I can see that actually being a win, or is it?

Reply:
Yes, I think a None, Some, Many with a custom iterator that iterates over all would be a fine abstraction here.